extensions/amp-script/amp-script.md

```diff
@@ -77,63 +109,103 @@ You can also include your JavaScript inline, in a `script` tag. You must:
 [/tip]
 
-## How does it work?
+### How does it work?
 
-`amp-script` runs your custom JavaScript in a [Web Worker](https://developer.mozilla.org/en-US/docs/Web/API/Web_Workers_API/Using_web_workers) that has access to a virtual DOM. When your JavaScript code modifies this virtual DOM, `amp-script` forwards these changes to the main thread and applies them to the `amp-script` element subtree.
+`amp-script` runs your custom JavaScript in a [Web Worker](https://developer.mozilla.org/en-US/docs/Web/API/Web_Workers_API/Using_web_workers) that has access to a virtual DOM. When the real DOM changes, `amp-script` updates the virtual DOM to match it. Similarly, when your JavaScript modifies the virtual DOM, `amp-script` updates the real DOM. In this way, the real DOM and virtual DOM are kept in sync.
 
-For example, adding an element to `document.body`:
+Under the hood, `amp-script` uses [@ampproject/worker-dom](https://github.com/ampproject/worker-dom/). For design details, see the ["Intent to Implement" issue](https://github.com/ampproject/amphtml/issues/13471).
 
-```js
-// my-script.js
-const p = document.createElement('p');
-p.textContent = 'I am added to the body!';
-document.body.appendChild(p);
-```
+## Capabilities
 
-Will be reflected on the page as a new child of the `amp-script` element:
+### Supported APIs
 
-```html
-<amp-script src="http://example.com/my-script.js" width="300" height="100">
-  <p>I am added to the body!</p>
-</amp-script>
+DOM elements and their properties are generally supported, with a few limits. For example, your code can't add a new `<script>` or `<style>` tag to the DOM.
+
+`amp-script` recreates many commonly used DOM APIs and makes them available to your code. This "hello, world" example uses `getElementByID()`, `addEventListener()`, `createElement()`, `textContent`, and `appendChild()`:
+
+```js
+const button = document.getElementById('textarea');
+
+button.addEventListener('click', () => {
+  const h1 = document.createElement('h1');
+  h1.textContent = 'Hello World!';
+  document.body.appendChild(h1);
+});
 ```
 
-Under the hood, `amp-script` uses [@ampproject/worker-dom](https://github.com/ampproject/worker-dom/). For design details, see the ["Intent to Implement" issue](https://github.com/ampproject/amphtml/issues/13471).
+Supported DOM APIs include:
 
-## State manipulation
+* Element getters like `getElementByName()`, `getElementsByClassName()`, `getElementsByTagName()`, `childNodes()`, `parentNode()`, and `lastChild()`
+* Mutators like `createTextNode()`, `appendChild()`, `insertBefore()`, `removeChild()`, and `replaceChild()`
+* Methods involving events like `addEventListener()`, `removeEventListener()`, and `createEvent()`
+* Property and attribute getters like `getAttribute()`, `hasAttribute()`
+* Event properties like `Event.target`, `Event.type`, and `Event.bubbles`
+* Element properties like `attributes`, `id`, `outerHTML`, `textContent`, `value`, `classList`, and `className`
+* And many more.
```

Comment:
Nit: Move the compat table link here.

Reply:
Moved it up here from a few paragraphs below.